app/code/Magento/Wishlist/Block/Customer/Wishlist.php

```diff
@@ -77,15 +89,67 @@ protected function _prepareCollection($collection)
         return $this;
     }
 
+    /**
+     * Retrieve Wishlist Product Items collection
+     *
+     * @return \Magento\Wishlist\Model\ResourceModel\Item\Collection
+     */
+    public function getWishlistItems()
+    {
+        $page = ($this->getRequest()->getParam("p")) ? $this->getRequest()->getParam("p") : 1;
+        $limit = ($this->getRequest()->getParam("limit")) ? $this->getRequest()->getParam("limit") : 10;
+        if ($this->_collection === null) {
```

Comment:
Please move the collection retrieval logic to separate i.e. "getCollection" method

Reply:
done